### PR TITLE
Add "DROP_ROWS" action to output of `OutliersDataCheck`

### LINKED ISSUE #2677: Add "DROP_ROW" action to ``OutliersDataCheck``
We should add an action to drop outlier rows in ``OutliersDataCheck``

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #2677

### CODE REVIEW

File: evalml/tests/data_checks_tests/test_outliers_data_check.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def test_outliers_data_check_warnings():
             ).to_dict()
         ],
         "errors": [],
-        "actions": [],
+        "actions": [{"code": "DROP_ROWS", "metadata": {"indices": [0, 3, 5, 10]}}],
```
Comment (chukarsten) [inline evalml/tests/data_checks_tests/test_outliers_data_check.py:40]: Can we add a second outlier in the same row, but different column?  I was going to write up an issue to increase coverage this way, but we could just do it here if you're game!  If not, let me know and I'll press Go on the issue.

Additionally, do we intend to handle that case with multiple outliers in the same row here?  Or keep the duplicate rows in DataCheckAction and let it handle that?

**Review by chukarsten — DISMISSED** [review]
It's clear that you started with clear code and a solid design as this was super easy to complete!  Very nice.  I'm really excited about Datacheck/actions and how it's progressing.

I think the only thing I'd like to see resolved would be the issue with duplicate rows going into the Datacheck.  That might already be determined so I'm going to go back through the issue now, so if it is, ping me and I'll re-review.

File: evalml/tests/data_checks_tests/test_outliers_data_check.py (hunk 1 above)
Comment (angela97lin) [reply evalml/tests/data_checks_tests/test_outliers_data_check.py:40]: That's a great idea! I think we should filter for duplicates, updated impl to handle this case and added a new test :)

**Review by chukarsten — APPROVED** [review]
Love it.

**Review by bchen1116 — APPROVED** [review]
Nice! LGTM

**Pull request merged**